### PR TITLE
Getting Started With Dependency Injection in Angular 12+

### LINKED ISSUE #3230: Getting Started With Dependency Injection in Angular 12+
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
Getting Started With Dependency Injection in Angular 12+

### Introduction paragraph (2-3 paragraphs):

Angular uses the Dependency Injection design pattern, which makes it extremely efficient. This programming paradigm allows classes, components, and modules to be interdependent while maintaining consistency. This reduces the frequency with which the class changes. 

In this tutorial, we will look at how Angular dependency injection works. The Angular creates a hierarchical dependency injection system. It creates a hierarchical tree of Injectors. Each Injector gets their own copy of Angular Providers. Together these two form the core of the Angular dependency injection framework. We will learn how Angular creates the injector tree. How injector resolves the dependency.
### Key takeaways: 

What is Dependency Injection?
Types of Dependency Injection in Angular
Advantages of Dependency Injection
The Drawbacks of not using Dependency Injection
Dependency Injection as a Design Pattern
Demo - Injecting Services into Components to Display a List
l.

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by NaomiSeint**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3230

### CODE REVIEW

**Comment by WanjaMIKE** [comment]
Plagiarism check done - Proceed. 
[Getting Started With Dependency Injection in Angular 12+ #3585.pdf](https://github.com/section-engineering-education/engineering-education/files/7072115/Getting.Started.With.Dependency.Injection.in.Angular.12%2B.3585.pdf)

File: content/articles/angular-12-dependency-injection/index.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
+### Introduction
+
```
Comment (jumamiller) [inline content/articles/angular-12-dependency-injection/index.md:2]: Hello @NaomiSeint , ensure that your article aligns with the guidlines set, kinldy restructure it well and remove all spaces where necessary as shown on this paragraph.

File: content/articles/angular-12-dependency-injection/index.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+  //getting the list of students and assigning the list to students array
+  getStudentsList() {
+    this.students = this.studentListService.getInternsDetails();
+  }
+}
+
+```
+In the above component class, we've defined the `students` array that we will use to hold the list of students. We then inject the ``StudentListService` that we previously created. This is how Dependency injection is achieved in Angular.
+
+We've also created the `getStudentsList()` method that we're using to get student details from the service. We then assign the results to the students array we've locally created. We call this method in the `ngOnInit()` method to get students' details on page load.
```
Comment (jumamiller) [inline content/articles/angular-12-dependency-injection/index.md:202]: There is a misplaced backtick on this paragraph, kindly check.

File: content/articles/angular-12-dependency-injection/index.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+</div>
+
+```
+
+Now run your Angular application.
+
+![Student table](/engineering-education/angular-di/output.png)
+
+### Conclusion
+In this tutorial, we've discussed a wide range of concepts of Angular DI with examples. We've seen how the services are created and then consumed within a component. We've also gone a step further to test our application on the browser.
```
Comment (jumamiller) [inline content/articles/angular-12-dependency-injection/index.md:244]: Conclusion should cover what has been covered on the entire article. Kindly fix.

**Review by jumamiller — CHANGES_REQUESTED** [review]
@NaomiSeint, there are minor suggestions I've made, kindly go over and rectify.Addtionally, I cant trace your author details.

**Comment by NaomiSeint** [comment]
@jumamiller ,all changes are done

**Comment by jumamiller** [comment]
Great work @NaomiSeint , finishing up tech review today!

**Comment by jumamiller** [comment]
Hello @NaomiSeint , could you add the author details before we move forward to final review.

**Comment by NaomiSeint** [comment]
Author details added.

**Review by jumamiller — APPROVED** [review]
A great way to kick off your articles,, looking forward for more improved articles. Great work @NaomiSeint!

**Comment by LinusMuema** [comment]
Hi @jumamiller 
The article does not have a hero image so please check on that.
My concern is that I am seeing some mistakes occurring for the third time this time. I had told you to [check my commit](https://github.com/section-engineering-education/engineering-education/pull/3312#issuecomment-907614928) for the changes but it seems you didn't. The same mistakes in the introduction and table of contents are there. 

Please make sure you check the CM's final edits to improve your review and prevent this from happening. 

cc: @hectorkambow advice on the same?

**Comment by jumamiller** [comment]
Hi @LinusMuema, I did check the commits and the changes were mainly on the spacings,  I'll be sure to look for a better a editor to improve on them.

**Comment by LinusMuema** [comment]
Awesome.
It's just the introductions and table of contents...

**Comment by LinusMuema** [comment]
Hi @NaomiSeint 
Thank you for joining section and we look forward to getting more amazing content from you. Awesome job!

cc: @jumamiller

**Pull request merged**